File: backend/JobTracker.Domain/Entities/Document.cs
namespace JobTracker.Domain.Entities;

public class Document
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid JobApplicationId { get; set; }
    public JobApplication? JobApplication { get; set; }
    public string OriginalFileName { get; set; } = string.Empty;
    public string StoredFileName { get; set; } = string.Empty;
    public string ContentType { get; set; } = string.Empty;
    public long SizeBytes { get; set; }
    public DocumentType Type { get; set; }
    public DateTime UploadedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/document.ts
export type DocumentType =
  | 0 // Resume
  | 1 // CoverLetter
  | 2 // JobDescription
  | 3 // TakeHomeAssignment
  | 4; // Other

export type Document = {
  id: string;
  jobApplicationId: string;
  originalFileName: string;
  contentType: string;
  sizeBytes: number;
  type: DocumentType;
  uploadedAt: string;
};
